src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -144,18 +145,19 @@ uct_cuda_copy_progress_event_queue(uct_cuda_copy_iface_t *iface,
     ucs_queue_iter_t iter;
 
     ucs_queue_for_each_safe(cuda_event, iter,
-                            &iface->outstanding_event_q[index], queue) {
+                            &iface->outstanding_event_q[id], queue) {
         result = cudaEventQuery(cuda_event->event);
         if (cudaSuccess != result) {
             break;
         }
-        ucs_queue_del_iter(&iface->outstanding_event_q[index], iter);
-        if (cuda_event->comp != NULL) {
-            uct_invoke_completion(cuda_event->comp, UCS_OK);
+        if (invoke_completions) {
+            ucs_queue_del_iter(&iface->outstanding_event_q[id], iter);
```

Comment:
inf loop

Reply:
Don't see why there would be inf loop but I've changed this part and leaving progress function unaltered. Instead I go through each element in the queues and return busy if there is an event that was successfully queried.